Cargo.toml

```diff
@@ -26,7 +26,7 @@ sha2 = "0.10"
 thiserror = "1.0"
 
 [dev-dependencies]
-criterion = "0.3"
+criterion = "0.5.1"
```

Comment:
Why specify all the way down to the rev?

Reply:
It's just a habit to take the latest version to define a range of acceptable versions. `0.5.1` means `[0.5.1, 0.6.0)` while `0.5` means `[0.5.0, 0.6.0)`. I don't see a reason to allow `0.5.0` if we know that `0.5.1` exist.